dfc/daemon.go

```diff
@@ -163,39 +160,68 @@ func (m *Smap) count() int {
 func (m *Smap) countProxies() int {
 	return len(m.Pmap)
 }
-func (m *Smap) countLocked() int {
-	m.lock()
-	defer m.unlock()
+
+func (m *Smap) countL() int {
+	smapLock.Lock()
+	defer smapLock.Unlock()
 	return m.count()
 }
 
 func (m *Smap) get(sid string) *daemonInfo {
-	si := m.Tmap[sid]
+	si, ok := m.Tmap[sid]
+	if !ok {
```

Comment:
This function can probably just be one line - i.e. `return m.Tmap[sid]`. See https://play.golang.org/p/cncYTcOVeyN. Same for `getProxy()` below.